### PR TITLE
add num_deletes in MvccProperties

### PR DESCRIPTION
**Pull request by hicqu**
###  What have you changed?

Add `num_deletes` into `MvccProperties`. It will make GC more accurate.
For compatibility, if a `MvccProperties`  is decoded from a engine which hasn't record `num_deletes`, it will be `num_versions - num_puts`.

###  What is the type of the changes?
Improvement.

###  How is the PR tested?
Manual test
- Create a TiDB cluster with TiKV master branch, insert some data into it.
- Delete some data, and then rolling update TiKV to the branch version.
- TiKV shouldn't panic even if it doesn't record `num_deletes`.
- Delete some data again, and use `tikv-ctl region-properties` to check it works as expected.

Here is a test example result:
For master branch TiKV:
图片
`num_versions` = `num_puts` + `num_deletes`.

For patched TiKV:
图片
`num_versions` > `num_puts` + `num_deletes`. It's because `num_rollbacks` is not included. So `num_deletes` is more accurate. It will be helpful for https://github.com/tikv/tikv/pull/6728/ .

###  Any examples? (optional)

### CODE REVIEW

File: components/raftstore/src/coprocessor/properties.rs [hunk 1]
```diff
@@ -76,6 +81,10 @@ impl MvccProperties {
         res.num_rows = props.decode_u64(PROP_NUM_ROWS)?;
         res.num_puts = props.decode_u64(PROP_NUM_PUTS)?;
         res.num_versions = props.decode_u64(PROP_NUM_VERSIONS)?;
+        // To be compatible with old versions.
+        res.num_deletes = props
+            .decode_u64(PROP_NUM_DELETES)
+            .unwrap_or_else(|_| res.num_versions - res.num_puts);
```
Comment (BusyJay) [inline components/raftstore/src/coprocessor/properties.rs]: If they are equal, then why add the metrics explicitly?

Comment (Little-Wallace) [reply components/raftstore/src/coprocessor/properties.rs]: I guess that it is just to be compatible with old versions....

Comment (hicqu) [reply components/raftstore/src/coprocessor/properties.rs]: They are not equal. Actually,
```
num_version = num_puts + num_rollback + num_lock + num_delete
```

Comment (Connor1996) [reply components/raftstore/src/coprocessor/properties.rs]: why not just put it to the last when encoding

**Pull request merged**